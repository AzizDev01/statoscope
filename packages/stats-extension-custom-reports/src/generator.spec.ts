import { ExtensionDescriptor } from '@statoscope/stats/spec/extension';
import Generator from './generator';

const adapter: ExtensionDescriptor = { name: 'test-adapter', version: '7.7.7' };

jest.mock('../package.json', () => ({
  ...jest.requireActual('../package.json'),
  version: '8.8.8',
}));

test('should work', () => {
  const generator = new Generator(adapter);

  generator.handleReport({
    context: 'context',
    data: 'data',
    id: 'foo',
    name: 'name',
    view: ['some-ui'],
  });

  generator.handleReport({
    context: 'context2',
    data: 'data2',
    id: 'foo2',
    name: 'name2',
    view: ['some-ui2'],
    compilation: 'foo-compilation',
    relations: [{ type: 'module', id: 'foo' }],
  });

  expect(generator.get()).toMatchSnapshot();
});

test('should not add existing report', () => {
  const generator = new Generator(adapter);

  generator.handleReport({
    context: 'context',
    data: 'data',
    id: 'foo',
    name: 'name',
    view: ['some-ui'],
  });

  generator.handleReport({
    context: 'context2',
    data: 'data2',
    id: 'foo',
    name: 'name2',
    view: ['some-ui2'],
  });

  expect(generator.get()).toMatchSnapshot();
});
