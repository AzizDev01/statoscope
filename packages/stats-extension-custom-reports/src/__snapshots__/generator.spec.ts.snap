// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `
Object {
  "descriptor": Object {
    "adapter": Object {
      "name": "test-adapter",
      "version": "7.7.7",
    },
    "author": Object {
      "name": "Sergey Melyukov",
      "url": "https://twitter.com/smelukov",
    },
    "description": "Statoscope extension to store custom reports in stats",
    "homepage": "https://github.com/statoscope/statoscope/",
    "name": "@statoscope/stats-extension-custom-reports",
    "version": "5.9.0",
  },
  "payload": Object {
    "compilations": Array [
      Object {
        "id": null,
        "reports": Array [
          Object {
            "context": "context",
            "data": "data",
            "id": "foo",
            "name": "name",
            "view": Array [
              "some-ui",
            ],
          },
        ],
      },
      Object {
        "id": "foo-compilation",
        "reports": Array [
          Object {
            "compilation": "foo-compilation",
            "context": "context2",
            "data": "data2",
            "id": "foo2",
            "name": "name2",
            "relations": Array [
              Object {
                "id": "foo",
                "type": "module",
              },
            ],
            "view": Array [
              "some-ui2",
            ],
          },
        ],
      },
    ],
  },
}
`;
